### PR TITLE
Add Integer#to_d

### PR DESCRIPTION
**Pull request by hachi8833**
Fixes https://github.com/goby-lang/goby/issues/647

### CODE REVIEW

File: vm/integer.go [hunk 1]
```diff
@@ -383,6 +383,21 @@ func builtinIntegerInstanceMethods() []*BuiltinMethodObject {
...
+		//
+		// ```Ruby
+		// 100.to_d # => '100'.to_d
+		// ```
+		// @return [Decimal]
+		{
+			Name: "to_d",
+			Fn: func(receiver Object, sourceLine int) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *normalCallFrame) Object {
+					r := receiver.(*IntegerObject)
```
Comment (st0012) [inline vm/integer.go]: Can you add argument check here?

Comment (hachi8833) [reply vm/integer.go]: I'll do that. 

I also found that arg checks are missing in other methods so I'd add them in another PR.

File: vm/integer.go [hunk 2]
```diff
@@ -383,6 +383,25 @@ func builtinIntegerInstanceMethods() []*BuiltinMethodObject {
...
+		//
+		// ```Ruby
+		// 100.to_d # => '100'.to_d
+		// ```
+		// @return [Decimal]
+		{
+			Name: "to_d",
+			Fn: func(receiver Object, sourceLine int) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *normalCallFrame) Object {
+					if len(args) > 1 {
```
Comment (st0012) [inline vm/integer.go]: why does it receive one argument? I don't see we use any here

Comment (hachi8833) [reply vm/integer.go]: I should've mentioned about https://github.com/goby-lang/goby/issues/648

Comment (st0012) [reply vm/integer.go]: I think either you also implement the feature that requires the argument, or you set the limit to 0 until you're going to implement that

**Pull request merged**